File: UI/careconnect_mobile/lib/models/messages/appointment_notification_message.dart
import 'package:careconnect_mobile/models/enums/appointment_status.dart';
import 'package:json_annotation/json_annotation.dart';

part 'appointment_notification_message.g.dart';

@JsonSerializable()
class AppointmentNotificationMessage {
  int appointmentId;
  int clientId;
  int employeeId;
  AppointmentStatus status;
  AppointmentStatus previousStatus;
  DateTime appointmentDate;
  String serviceName;
  DateTime changedAt;
  String changedBy;

  AppointmentNotificationMessage({
    required this.appointmentId,
    required this.clientId,
    required this.employeeId,
    required this.status,
    required this.previousStatus,
    required this.appointmentDate,
    required this.serviceName,
    required this.changedAt,
    required this.changedBy,
  });

  factory AppointmentNotificationMessage.fromJson(Map<String, dynamic> json) =>
      _$AppointmentNotificationMessageFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentNotificationMessageToJson(this);
}

File: CareConnect/CareConnect.Models/Messages/AppointmentNotificationMessage.cs
using CareConnect.Models.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace CareConnect.Models.Messages
{
    public class AppointmentNotificationMessage
    {
        public int AppointmentId { get; set; }

        public int ClientId { get; set; }

        public int EmployeeId { get; set; }

        public AppointmentStatus Status { get; set; }

        public DateTime AppointmentDate { get; set; }

        public string ServiceName { get; set; }

        public DateTime ChangedAt { get; set; }
    }
}
